src/OrleansHost/WindowsServerHost.cs

```diff
@@ -114,7 +108,8 @@ private int RunImpl(CancellationToken? cancellationToken = null)
         public bool ParseArguments(string[] args)
         {
             string siloName = Dns.GetHostName(); // Default to machine name
-            SiloHost = new SiloHost(siloName);
+            var builder = new SiloHostBuilder();
+            SiloHost = builder.Build(siloName);
 
```

Comment:
This is confusing to me. Initialization of siloHost should be in its own method, maybe called initSiloHost(args), it shouldn't be inside ParseArguments function. ParseAtguments func should be responsible just for parse args from command line params.